### PR TITLE
Dev/robin/8859 simplehash schema v3

### PR DESCRIPTION
**Pull request by robinbryce**
feat: v3 schema for forestrie

This new schema is required in order to ommit confirmation_status from
the hash. For simplehash assets, this field was crucial. For forestrie
it is only informative. And crucially, it changes after the event is
first added to the log STORED -> COMMITTED -> CONFIRMED -> UNEQUIVOCAL

We took the opportunity to prune other un-necessary fields:

* asset_identity is redundant and will soon be confusing.
* from is not relevant for forestrie (it was the custodial wallet
  address for khipu)

Note that idtimestamp is NOT added. The details of the leaf hash
construction are governed by forestrie and only change if the merkle log
format version is incremented. And the database can't store the log
values in native format anyway.

Additionally,

ci: enable code quality and unit tests on ci

### CODE REVIEW

File: simplehash/hasher.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+func NewEventMarshaler() *simpleoneof.Marshaler {
+	return v2assets.NewFlatMarshalerForEvents()
+}
+
+func (h *Hasher) applyEventOptions(o HashOptions, event *v2assets.EventResponse) {
+	if o.publicFromPermissioned {
+		PublicFromPermissionedEvent(event)
+	}
+
+	// If the caller is responsible for evidence confirmation they will have a
```
Comment (honourfish) [inline simplehash/hasher.go]: no longer need this?

I would even advocate to have the option removed, we should try and avoid changing the state of the event properties before hashing?

File: simplehash/schemav3.go [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+	"encoding/json"
+	"fmt"
+	"hash"
+
+	v2assets "github.com/datatrails/go-datatrails-common-api-gen/assets/v2/assets"
+	"github.com/datatrails/go-datatrails-common-api-gen/marshalers/simpleoneof"
+	"github.com/zeebo/bencode"
+)
+
+// V3Event is a struct that contains ONLY the event fields we want to hash for schema v2
```
Comment (honourfish) [inline simplehash/schemav3.go]:
```suggestion
// V3Event is a struct that contains ONLY the event fields we want to hash for schema v3
```

File: simplehash/schemav3.go [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+		return V3Event{}, err
+	}
+	return V3FromEventJSON(eventJson)
+}
+
+// HashEvent hashes a single event according to the canonical simple hash event
+// format available to api consumers. The source event is in the grpc proto buf
+// format. GRPC endpoints are not presently exposed by the platform.
+//
+// Options:
```
Comment (honourfish) [inline simplehash/schemav3.go:94]: nice!

File: simplehash/hasher.go (hunk 1 above)
Comment (robinbryce) [reply simplehash/hasher.go]: which one(s) its not clear from the diff ? but yes in principal

File: simplehash/schemav2.go [hunk 4]
```diff
@@ -50,45 +57,19 @@ func (h *HasherV2) Reset() { h.hasher.Reset() }
...
 //     permissioned (owner) counter part of a public attestation.
-//   - WithAsConfirmed should be set if the caller is implementing CONFIRMATION
-//     as part of an evidence subsystem implementation. The expectation is that
-//     the caller has a PENDING record to hand, and is in the process of
-//     creating the CONFIRMED record. It is the CONFIRMED record that needs to
+//   - WithAsCommitted should be set if the caller is implementing COMMITMENT
+//     trust level as part of an evidence subsystem implementation. The
+//     expectation is that the caller has a STORED record to hand, and is in
+//     the process of creating the COMMITTED record. It is the COMMITTED record
+//     that needs to
```
Comment (honourfish) [inline simplehash/schemav2.go]:
```suggestion
```

**Review by honourfish — APPROVED** [review]

File: simplehash/hasher.go (hunk 1 above)
Comment (honourfish) [reply simplehash/hasher.go]: Ahh sorry this was asCommitted, which you have now removed :)

**Pull request merged**